src/components/Pensjonssimulering/Pensjonssimulering.tsx

```diff
@@ -58,58 +86,92 @@ export function Pensjonssimulering({ uttaksalder }: PensjonssimuleringProps) {
+      startMaaned: startMaaned,
+      uttaksgrad: uttaksgrad,
+    })
+    setAlderspensjonRequestBody(requestBody)
+  }, [afp, person, startAlder, startMaaned, uttaksgrad])
+
+  useEffect(() => {
+    if (startAlder) {
+      const aarArray = generateXAxis(startAlder, MAX_UTTAKSALDER)
+      setChartOptions({
+        chart: {
           type: 'column',
-          pointWidth: COLUMN_WIDTH,
-          name: SERIE_NAME_AFP,
-          color: SERIE_COLOR_AFP,
-          data: simulateDataArray(AFP_DATA, aarArray.length),
+          scrollablePlotArea: {
+            minWidth: aarArray.length * COLUMN_WIDTH * 1.6,
+            scrollPositionX: 0,
+          },
         },
-        {
-          type: 'column',
-          pointWidth: COLUMN_WIDTH,
-          name: SERIE_NAME_TP,
-          color: SERIE_COLOR_TP,
-          data: simulateTjenestepensjon(uttaksalder, MAX_UTTAKSALDER),
+        xAxis: {
+          categories: aarArray,
         },
-        {
-          type: 'column',
-          pointWidth: COLUMN_WIDTH,
-          name: SERIE_NAME_ALDERSPENSJON,
-          color: SERIE_COLOR_ALDERSPENSJON,
-          data: simulateDataArray(
-            FOLKETRYGDEN_DATA,
-            aarArray.length,
-            uttaksalder,
-            18_000
-          ),
-        },
-      ],
-    })
-  }, [uttaksalder])
+        series: [
+          {
+            type: 'column',
+            pointWidth: COLUMN_WIDTH,
+            name: SERIE_NAME_INNTEKT,
+            color: SERIE_COLOR_INNTEKT,
+            data: [...PENSJONSGIVENDE_DATA].splice(0, aarArray.length),
+          },
+          ...(afp === 'ja_privat'
+            ? [
+                {
+                  type: 'column',
+                  pointWidth: COLUMN_WIDTH,
+                  name: SERIE_NAME_AFP,
+                  color: SERIE_COLOR_AFP,
+                  /* c8 ignore next 1 */
+                  data: processPensjonsberegningArray(alderspensjon?.afpPrivat),
+                } as SeriesOptionsType,
+              ]
+            : []),
+          ...(harSamtykket
+            ? [
+                {
+                  type: 'column',
+                  pointWidth: COLUMN_WIDTH,
+                  name: SERIE_NAME_TP,
+                  color: SERIE_COLOR_TP,
+                  /* c8 ignore next 1 */
+                  data: simulateTjenestepensjon(startAlder, MAX_UTTAKSALDER),
+                } as SeriesOptionsType,
+              ]
+            : []),
+          {
+            type: 'column',
+            pointWidth: COLUMN_WIDTH,
+            name: SERIE_NAME_ALDERSPENSJON,
+            color: SERIE_COLOR_ALDERSPENSJON,
+            data: processPensjonsberegningArray(alderspensjon?.alderspensjon),
+          },
+        ],
+      })
+    }
+  }, [startAlder, alderspensjon])
 
   return (
     <section className={styles.section}>
+      {isLoading && (
+        <Loader
+          data-testid="loader"
+          size="3xlarge"
+          title="Et øyeblikk, vi simulerer pensjonen din"
```

Comment:
Vet ikke om vi skal bruke begrepet "simulerer" for visning til brukeren. Forslag: "beregner".